test/init_test.go

```diff
@@ -44,7 +44,7 @@ import (
 
 var initMetrics sync.Once
 
-func setup(t *testing.T) (*clients, string) {
+func setup(t *testing.T, fn ...func(*testing.T, *clients, string)) (*clients, string) {
```

Comment:
Nice :)

I wonder if we should do something similar in the tearDown?
I tests are executed in a disposable environment it may not be so critical, but still we may have multiple tests setting up the registry and colliding with each other.

Reply:
@afrittoli depends on what you mean by colliding with each others ? :thinking: `setup` is ran for each `Test*`, and it creates a new namespace, so each element that calls `setup(…)` is isolated from each others.
And the cleaning part removes any resource and the namespace so no real need for additionnal `tearDown` thing here

Reply:
Oh, that's a good point, thank you